### DISCUSSION TITLE
onChange event must return details about what has changed.

### DISCUSSION THREAD

**Discussion by dilipmer** [iddan/react-spreadsheet#305] [Ideas]
Currently onChage event only return updated matrices. it should also provide more details on what has changed. like provide coords of cell which is changed.

onCellCommit doesn't work properly it just keeps once I make a change.

**Comment by dilipmer** [comment]
Hello there any update in here?

**Comment by iddan** [comment]
Hey, I'm open for PRs. Currently, I can't find the capacity to implement this but I'm willing to review PRs

**Comment by hadri3n** [comment]
@dilipmer Why do you want to do that ? What is your use case ? 
Because you can retrieve the current active cell coordinates (the one being modified) pretty easily using the `onActivate` prop function.

Short example displaying the currently edited value beneath the spreadsheet:
```javascript
const App = () => {
  const [data, setData] = useState<Matrix<CellBase>>([
    [{ value: "Vanilla" }, { value: "Chocolate" }],
    [{ value: "Strawberry" }, { value: "Cookies" }],
  ]);
  const [currentActive, setCurrentActive] = useState<Point | undefined>();

  return (
    <div>
      <div>
        <Spreadsheet
          data={data}
          onChange={setData}
          onActivate={setCurrentActive}
        />
      </div>
      <div>
        <pre>{JSON.stringify(data)}</pre>
        <pre>{JSON.stringify(currentActive)}</pre>
        <pre>Currently edited value:
          {JSON.stringify(
            data[currentActive?.row || 0][currentActive?.column || 0]
          )}
        </pre>
      </div>
    </div>
  );
};
```
